org.gecko.emf.exporter/src/org/gecko/emf/exporter/AbstractEMFExporter.java

```diff
@@ -0,0 +1,1957 @@
+		resetStopwatch();
+
+		logger.info("Starting preprocessing EObjects contained in Resource");
+
+		final PreProcessedResourceEObjectsDTO preProcessedEObjectsDTO = new PreProcessedResourceEObjectsDTO();
+
+		preProcessedEObjectsDTO.rawResourceEObjects.addAll(resource.getContents());
+
+		preProcessedEObjectsDTO.resourceEObjectsURIs
+				.addAll(extractResourceEObjectsURIs(preProcessedEObjectsDTO.rawResourceEObjects));
+
+		final Set<String> processedEObjectsIdentifiers = new HashSet<String>();
+
+		preProcessResourceEObjects(preProcessedEObjectsDTO.rawResourceEObjects, processedEObjectsIdentifiers,
+				preProcessedEObjectsDTO, exportOptions);
+
+		if (!preProcessedEObjectsDTO.unProcessableResourceEObjects.isEmpty()) {
+			EcoreUtil.deleteAll(preProcessedEObjectsDTO.unProcessableResourceEObjects, true);
+		}
+
+		logger.info("Finished preprocessing EObjects contained in Resource in {} second(s)", elapsedTimeInSeconds());
+
+		return preProcessedEObjectsDTO.preProcessedResourceEObjects;
+	}
+
+	private List<URI> extractResourceEObjectsURIs(List<EObject> resourceEObjects) {
+		return resourceEObjects.stream().map(eObject -> EcoreUtil.getURI(eObject)).collect(Collectors.toList());
+	}
+
+	private void preProcessResourceEObjects(List<? extends EObject> eObjects, Set<String> processedEObjectsIdentifiers,
+			PreProcessedResourceEObjectsDTO preProcessedEObjectsDTO, Map<Object, Object> exportOptions) {
+		for (EObject eObject : eObjects) {
+			if (isProcessed(processedEObjectsIdentifiers, eObject)) {
+				continue;
+			}
+
+			processedEObjectsIdentifiers.add(getEObjectIdentifier(eObject));
+
+			preProcessResourceEObject(eObject, preProcessedEObjectsDTO);
+
+			for (EReference eReference : eObject.eClass().getEAllReferences()) {
+				preProcessResourceEObject(eObject, eReference, processedEObjectsIdentifiers, preProcessedEObjectsDTO,
+						exportOptions);
+			}
+		}
+	}
+
+	@SuppressWarnings("unchecked")
+	private void preProcessResourceEObject(EObject eObject, EReference eReference,
+			Set<String> processedEObjectsIdentifiers, PreProcessedResourceEObjectsDTO preProcessedEObjectsDTO,
+			Map<Object, Object> exportOptions) {
+
+		Object rawEObject = eObject.eGet(eReference, exportNonContainmentEnabled(exportOptions));
+
+		if (rawEObject != null) {
+
+			if (!eReference.isMany()) {
+
+				if (shouldPreProcessResourceEObject((EObject) rawEObject, preProcessedEObjectsDTO)) {
+					preProcessResourceEObject((EObject) rawEObject, preProcessedEObjectsDTO);
+				} else {
+					preProcessedEObjectsDTO.unProcessableResourceEObjects.add((EObject) rawEObject);
+				}
+
+			} else if (eReference.isMany()) {
+
+				if (((List<EObject>) rawEObject).isEmpty()) {
+					EcoreUtil.remove(eObject, eReference, rawEObject);
+					return;
+				}
+
+				List<EObject> processableEObjects = new ArrayList<>();
+				List<EObject> unProcessableEObjects = new ArrayList<>();
+
+				for (EObject maybeProcessableEObject : (List<EObject>) rawEObject) {
```

Comment:
Please use the following way to iterate over this list:

```java
for (int i = 0; i < list.size(); i++) {
    EObject eObject = list.basicGet(i);
    //This will prevent any proxy to be resolved. Usually the list will be an EcoreEList, where the get(i) causes the proxy to beeing resolved (aka loaded e.g. from the database)
}

```